### PR TITLE
typstyle: init at 0.11.5

### PR DESCRIPTION
**Pull request by drupol**
CC: @Enter-Tainer

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ty/typstyle/package.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  cargoHash = "sha256-oLJWgF5byM3sY3Bs/wpSrBqjNg4sHDF3RIsWZBiguGI=";
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  buildInputs = [
+    libgit2
+    openssl
```
Comment (Enter-tainer) [inline pkgs/by-name/ty/typstyle/package.nix:31]: does it need openssl? i dont really know. maybe it is test dependency

**Review by Enter-tainer — COMMENTED** [review]
thank you!

File: pkgs/by-name/ty/typstyle/package.nix (hunk 1 above)
Comment (Enter-tainer) [reply pkgs/by-name/ty/typstyle/package.nix:31]: other part looks good to me. typstyle is a typical rust cli tool and doesnt have complex build logic👍

**Comment by drupol** [comment]
I just have this while compiling:

```
typstyle>    Compiling xattr v1.3.1
typstyle> warning: typstyle@0.11.5: could not find repository at '/build/source'; class=Repository (6); code=NotFound (-3)
typstyle> warning: typstyle@0.11.5: VERGEN_GIT_DESCRIBE set to default
typstyle> warning: typstyle@0.11.5: VERGEN_GIT_SHA set to default
typstyle>    Compiling wasm-bindgen v0.2.92
```

Maybe there's a way to specify the version when compiling ?

File: pkgs/by-name/ty/typstyle/package.nix (hunk 1 above)
Comment (drupol) [reply pkgs/by-name/ty/typstyle/package.nix]: You were right, openssl was not required, I removed it.

**Comment by Enter-tainer** [comment]
`typstyle --version` includes weird data but I think `typstyle -V` should work as expected. `-V` is the short version string and `--version` is the "long version string".

The long version string is generated by reading time/git-describe/cargo-triple and they are not available when reproducible build is required.

**Comment by Enter-tainer** [comment]
I can optimize this by reading version from better place in next release and `--version` will become better

**Comment by drupol** [comment]
Nix builds software in total isolation, even isolated from the Git files. Therefore, relying on something else than Git is a good idea. So yeah, maybe reading it from `cargo.toml` would be a better choice.

**Comment by drupol** [comment]
Pushing missing package for building on Darwin.

**Pull request merged**